### PR TITLE
Local docker

### PR DESCRIPTION
**Pull request by rdubrock**
Provides a script and some configuration to run grafana with provisioning in a docker container

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      - The `grafanaName` needs to exactly match the names specified in your datasource provisioning
+      - The `hostedId` is the same value as the `basicAuthUser` in your datasource provisioning
+      - `publisherToken` needs to be a grafana.com api key with a `MetricsPublisher` role. This is what the probes use to publish metrics to your cloud stack.
+
+Grafana configuration can be adjusted using the `custom.ini` file located in `/scripts`. It defaults to development app mode, and has some feature toggles. Grafana will need to be restarted to pick up changes.
+
+### Run
+
+- Run `yarn watch`
+- In a seperate terminal window, run `./scripts/run-local`. You can pass an optional `-v` flag to pick a specific version of Grafana to run. i.e. `./scripts/run-local -v 8.0.1`
```
Comment (electron0zero) [inline CONTRIBUTING.md:41]:
```suggestion
- In a separate terminal window, run `./scripts/run-local.sh`. You can pass an optional `-v` flag to pick a specific version of Grafana to run. i.e. `./scripts/run-local.sh -v 8.0.1`
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+- Go to `localhost:3000`
+- Changes to the plugin code require refreshing the browser. Changes to provisioning require restarting Grafana (which will happen if you just rerun the `run-local` script).
```
Comment (electron0zero) [inline CONTRIBUTING.md:43]:
```suggestion
- Changes to the plugin code require refreshing the browser. Changes to provisioning require restarting Grafana (which will happen if you just rerun the `run-local.sh` script).
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+### Running the entire stack locally
+
+- See the instructions for [setting up the api](https://github.com/grafana/synthetic-monitoring-api/blob/main/DEVELOPMENT.md)
```
Comment (electron0zero) [inline CONTRIBUTING.md:47]: this link is private :)

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
+# Contributing to this repository
+
+## Getting started
+
+There are two different "modes" for developing this plugin. You can either work on just the plugin, pointed at Grafana cloud (easier to set up), or if you need to work across the entire stack you'll need to set up all the dependent components.
+
+### Dependencies
```
Comment (electron0zero) [inline CONTRIBUTING.md:7]: we need SQLite as well, for grafana database.

File: scripts/custom.ini [hunk 5]
```diff
@@ -0,0 +1,816 @@
...
+;oauth_auto_login = false
+
+# OAuth state max age cookie duration in seconds. Defaults to 600 seconds.
+;oauth_state_cookie_max_age = 600
+
+# limit of api_key seconds to live before expiration
+;api_key_max_seconds_to_live = -1
+
+#################################### Anonymous Auth ######################
+[auth.anonymous]
```
Comment (electron0zero) [inline scripts/custom.ini:310]: let's enable this and drop users on home page so we don't have to login each time.

File: scripts/custom.ini [hunk 6]
```diff
@@ -0,0 +1,816 @@
...
+# enable anonymous access
+;enabled = false
+
+# specify organization name that should be used for unauthenticated users
+;org_name = Main Org.
+
+# specify role for unauthenticated users
+;org_role = Viewer
```
Comment (electron0zero) [inline scripts/custom.ini:318]:
```suggestion
# enable anonymous access
enabled = true

# specify organization name that should be used for unauthenticated users
org_name = Main Org.

# specify role for unauthenticated users
org_role = Admin
```

**Review by electron0zero — APPROVED** [review]
lgtm, with few suggestions 👍🏼

File: CONTRIBUTING.md (hunk 4 above)
Comment (rdubrock) [reply CONTRIBUTING.md:7]: I think that comes with the docker image

File: CONTRIBUTING.md (hunk 3 above)
Comment (rdubrock) [reply CONTRIBUTING.md:47]: I know, but you have to have access to that repo in order to run the whole stack locally

**Pull request merged**